src/pudl/analysis/plant_parts_eia.py

```diff
@@ -1567,6 +1560,50 @@ def execute(self, part_df, gens_mega):
         return part_df
 
 
+class AddMaxMinAttribute(AddAttribute):
+    """
+    Add Attributes based on the maximum or minimum value of a sorted attribute.
+
+    This object adds an attribute based on the maximum or minimum of another
+    attribute within a group of plant parts uniquely identified by their base
+    ID columns.
+    """
+
+    # should this function be able to add a max and min attribute at the same time?
+    # e.g. construction_year and installation_year at the same time
+    # is that actually more efficient?
+    def execute(self, part_df, gens_mega):
+        """
+        Add the attribute to the plant part df based on sorting of another attribute.
+
+        Args:
+            part_df (pandas.DataFrame): dataframe containing records associated
+                with one plant part.
+            gens_mega (pandas.DataFrame): a table of all of the generators with
+                identifying columns and data columns, sliced by ownership which
+                makes "total" and "owned" records for each generator owner.
+        """
+        attribute_col = self.attribute_col
+        if attribute_col in part_df.columns:
+            logger.debug(f"{attribute_col} already here.. ")
+            return part_df
+
+        logger.debug(f"pre count of part DataFrame: {len(part_df)}")
+        attribute_params = MAX_MIN_ATTRIBUTES_DICT[attribute_col]
```

Comment:
I think this step might feel more straightforward if these `attribute_params` were actually passed in as args. It would make it more clear in the docstring/args what the options are.